jax/_src/numpy/lax_numpy.py

```diff
@@ -3866,6 +3882,8 @@ def _eliminate_deprecated_list_indexing(idx):
   # non-tuple sequence containing slice objects, [Ellipses, or newaxis
   # objects]". Detects this and raises a TypeError.
   if not isinstance(idx, tuple):
+    if isinstance(idx, (int, slice)):
+      return (idx,)
```

Comment:
This is a tiny improvement to early exit in case of integer indices. `isinstance(idx, typing.Sequence)` is quite slow since it calls `__instancecheck__`. `isinstance(idx, collections.abc.Sequence)` would be faster in Python 3.9, but early exiting for the simple case shouldn't complicate things too much here.